assets/src/hooks/v2/use_api_response.tsx

```diff
@@ -147,29 +151,43 @@ const useApiPath = (screenId: string, appendPath?: string): string => {
       ...loggingParams(),
     };
 
-    for (const [key, value] of Object.entries(params)) {
+    if (queryParams) {
+      queryParams.forEach((value, key) => {
+        url.searchParams.append(key, value);
+      });
+    }
+
+    for (const [key, value] of Object.entries(datasetParams)) {
       if (value) url.searchParams.append(key, value);
     }
 
     return url.toString();
   }, [screenId, appendPath]);
 };
 
+interface UseBaseApiResponseProps {
```

Comment:
nit: These aren't "props" as `useBaseApiResponse` is not a React component. `UseApiResponseOpts`?